### PR TITLE
Fix some todo's

### PR DESCRIPTION
**Pull request by danewalton**
- remove "hubClient" naming from adu pieces
- remove old comments for todo's which were patched
- use the "accept" or "reject" parameter for the payload response

### CODE REVIEW

File: libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c [hunk 1]
```diff
@@ -429,18 +425,18 @@ static void prvFillBaseAduInstallResults( AzureIoTHubClientADUInstallResult_t *
...
-                                                   AzureIoTHubClientADUInstallResult_t * pxUpdateResults,
+                                                   AzureIoTADUClientInstallResult_t * pxUpdateResults,
                                                    uint8_t * pucBuffer,
                                                    uint32_t ulBufferSize,
                                                    uint32_t * pulRequestId )
 {
     az_result xAzResult;
     az_iot_adu_client_device_information xBaseDeviceInformation;
     az_iot_adu_client_workflow xBaseWorkflow;
-    az_iot_adu_client_install_result xInstallResult; /* TODO: fill up. */
```
[the remark this answers is not in the document]
Comment (ewertons) [reply libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c:443]: So prvFillBaseAduInstallResults actually "translates" the ADU results from the FreeRTOS middleware to the azure-sdk-for-c layer, but the results themselves never get filled up in xInstallResult (up to my knowledge, unless you implemented that), so the TODO still applies.

File: libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c [hunk 2]
```diff
@@ -311,18 +309,16 @@ AzureIoTResult_t AzureIoTADUClient_SendResponse( AzureIoTADUClient_t * pxAzureIo
...
     }
 
-    /* TODO: consume xRequestDecision in az_iot_adu_client_get_service_properties_response */
-    ( void ) xRequestDecision;
+    /* Code 406 is "Not Acceptable" */
     xAzResult = az_iot_adu_client_get_service_properties_response(
         &pxAzureIoTADUClient->_internal.xADUClient,
         ( int32_t ) ulPropertyVersion,
-        200,
+        xRequestDecision == eAzureIoTADURequestDecisionAccept ? 200 : 406,
```
Comment (ewertons) [inline libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c:316]: What do you think about defining this values (200, 406) as macros at the top, with some self-explanatory names?

Suggestion:

```c
#define AZURE_IOT_ADU_REQUEST_ACCEPTED 200
#define AZURE_IOT_ADU_REQUEST_REJECTED 406
```

**Review by ewertons — APPROVED** [review]
Approved with comments.

File: libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c (hunk 2 above)
Comment (danewalton) [reply libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c:316]: yes yes definitely

File: libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (danewalton) [reply libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c:443]: okay restoring the todo for now to look into it

**Pull request merged**